lib/request.ts

```diff
@@ -48,11 +100,9 @@ export function getRequest({
 		  }
 		: utils.debugRequest;
 
-	const exports = {};
-
-	const prepareOptions = async function (options) {
+	const prepareOptions = async function (options: BalenaRequestOptions) {
 		if (options == null) {
-			options = {};
+			options = {} as BalenaRequestOptions;
```

Comment:
This should not need to be cast, if `options` is actually optional and therefore every property is actually optional then the types should reflect that and not need casting. In fairness though I wouldn't be surprised if it's actually required and this defaulting should be removed

Reply:
So as part of this commit I just converted to TS, but I agree on removing all such defaulting so that we can get rid of this weird casting.